src/matlab2tikz.m

```diff
@@ -2267,12 +2294,19 @@ function assertRegularAxes(handle)
     % that is set when the color == 'none' (if it is omitted, the property will not
     % be set).
     % TODO: probably this should be integrated with getAndCheckDefault etc.
+    if opts_has(options,property) && isNone(opts_get(options,property))
+        return
+    end
     if ~isNone(color)
         [m2t, xcolor] = getColor(m2t, handle, color, 'patch');
         if ~isempty(xcolor)
             % this may happen when color == 'flat' and CData is Nx3, e.g. in
             % scatter plot or in patches
-            options = opts_add(options, property, xcolor);
+            if isempty(property)
+                options = opts_add(options, xcolor);
+            else
+                options = opts_add(options, property, xcolor);
+            end
```

Comment:
@patrickkwang  Exactly. :+1:
@miscco Not that I know of. Note that this is within some function related to colors where one of the arguments was `key` in `key=value`. The patch makes sure that we can signal that we need just `value` in the TikZ output.